Sources/MySQL/Connection/MySQLGeometry.swift

```diff
@@ -0,0 +1,92 @@
+public enum MySQLGeometry: Equatable {
+    case point(x: Double, y: Double)
+
+    fileprivate var wkbType: UInt32 {
+        switch self {
+        case .point: return 1
+        }
+    }
+
+    fileprivate static let headerSize = 9
+}
+
+extension MySQLGeometry: CustomStringConvertible {
+    public var description: String {
+        switch self {
+        case .point(x: let x, y: let y):
+            return "Point(\(x) \(y))"
+        }
+    }
+}
+
+extension MySQLGeometry {
+    func convertToData() -> Data {
+        switch self {
+        case .point(let x, let y):
+            let bufferSize = MySQLGeometry.headerSize + 2 * MemoryLayout<Double>.size
+            var buffer = ByteBufferAllocator().buffer(capacity: bufferSize)
+            buffer.write(integer: UInt32(0)) // 4 byte filler
+            buffer.write(integer: UInt8(1)) // set little endian byte order
+            buffer.write(integer: wkbType, endianness: .little) // geometry type
+            buffer.write(floatingPoint: x)
+            buffer.write(floatingPoint: y)
+            guard let data = buffer.getData(at: 0, length: bufferSize) else {
+                fatalError("Could not create Data from Buffer.")
+            }
+            return data
+        }
+    }
+
+    static func convertFromData(_ data: Data) throws -> MySQLGeometry {
+        let count = data.count
+        guard count >= headerSize else {
+            throw MySQLError(identifier: "geometryType", reason: "Not enough data to read geometry type.")
+        }
+        var buffer = ByteBufferAllocator().buffer(capacity: count)
+        buffer.withUnsafeMutableWritableBytes { (p) in
+            p.copyBytes(from: data)
+        }
+
+        let _: UInt32 = try buffer.requireInteger()
+        let _: UInt8 = try buffer.requireInteger()
```

Comment:
I don't know if it's possible, but you may want to assert that this is little-endian.

Reply:
Will do